lambda_function.py

```diff
@@ -1,9 +1,161 @@
+from typing import Dict
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class Onboarder(object):
+
+    MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME = 'Dome9AutomaticOnboardigStackSet'
+    CUSTOMER_ACCOUNT_EXECUTION_ROLE_NAME = "AWSControlTowerExecution"
+    MASTER_ACCOUNT_STACK_SET_ROLE = "service-role/AWSControlTowerStackSetRole"
+    STACK_OPERATION_WAIT_RETRIES = 12
+    STACK_OPERATION_WAIT_SLEEP = 10
+
+    def __init__(self, region_name: str, customer_account_id: str, customer_account_name: str, readonly: bool = True) -> None:
+        logger.info(f"Initing onboarder with region_name: '{region_name}', customer_account_id: '{customer_account_id}', customer_account_name: '{customer_account_name}'")
+        self.region_name = region_name
+        self.customer_account_id = customer_account_id
+        self.master_account_id = self.retrieve_master_account_id()
+
+        self.customer_account_name = customer_account_name
+        self.customer_account_new_role_name = f"Dome9Role-{self.master_account_id}-{self.customer_account_id}"
+        self.customer_account_new_role_arn = f"arn:aws:iam::{self.customer_account_id}:role/{self.customer_account_new_role_name}"
+        self.customer_account_external_id = self.generate_external_id()
+
+        self.cloudformation_client = boto3.client("cloudformation")
+        if readonly:
+            self.user_side_stack_cf_filename = "user_side_stack.yaml"
+        else:
+            self.user_side_stack_cf_filename = "user_side_stack.yaml"
+
+    @staticmethod
+    def retrieve_master_account_id() -> str:
+        sts_client = boto3.client('sts')
+        response = sts_client.get_caller_identity()
+        return response.get("Account")
+
+    @staticmethod
+    def generate_external_id() -> str:
+        return str(uuid4())[:8]
+
+    def create_stack_set(self) -> Dict:
+        administrator_role_arn = f"arn:aws:iam::{self.master_account_id}:role/{self.MASTER_ACCOUNT_STACK_SET_ROLE}"
+
+        logger.info(f"Creating StackSet: {self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME}, "
+                    f"administrator_role_arn : {administrator_role_arn}, "
+                    f"customer_account_external_id: {self.customer_account_external_id}, "
+                    f"customer_account_new_role_name: {self.customer_account_new_role_name}")
+
+        with open(self.user_side_stack_cf_filename) as f:
+            template_body = f.read()
+
+        response = self.cloudformation_client.create_stack_set(
+            StackSetName=Onboarder.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME,
+            Description='Dome9 auto onboarding stack set',
+            TemplateBody=template_body,
+            Parameters=[{'ParameterKey': 'Externalid', 'ParameterValue': "Placeholder"},
+                        {'ParameterKey': 'AccountRoleName', 'ParameterValue': "Placeholder"}],
+            Capabilities=[
+                'CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM', 'CAPABILITY_AUTO_EXPAND'
+            ],
```

Comment:
you should run our `yapf` to align the code styling

Reply:
done